arch/arm/src/imxrt/hardware/imxrt_flexio.h

```diff
@@ -0,0 +1,677 @@
+/****************************************************************************
+ * arch/arm/src/imxrt/hardware/imxrt_flexio.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_ARM_SRC_IMXRT_HARDWARE_IMXRT_FLEXIO_H
+#define __ARCH_ARM_SRC_IMXRT_HARDWARE_IMXRT_FLEXIO_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include "hardware/imxrt_memorymap.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+struct flexio_type_s
+{
+  volatile uint32_t VERID;            /* Version ID Register, offset: 0x0 */
```

Comment:
I would rather support to use defines for register offsets and get/put/modify APIs to access the value, but this approach is also fine. One thing that might give better readability is to use uint32_t arrays to fill the gaps.

Reply:
@pkarashchenko It is way easier to define register structure this way, especially if it is a big one. It is also easier to use.

Reply:
Yes, but in a way you used here usually r/w attributes are added also. Like `volatile const uint32_t TIMSTAT` instead of `volatile uint32_t TIMSTAT` to specify RO permissions :)